pkg/controller/dns/external_dns_reconciler.go

```diff
@@ -0,0 +1,37 @@
+package dns
+
+import (
+	"time"
+
+	"github.com/Azure/aks-app-routing-operator/pkg/config"
+	"github.com/Azure/aks-app-routing-operator/pkg/controller/common"
+	"github.com/Azure/aks-app-routing-operator/pkg/manifests"
+	appsv1 "k8s.io/api/apps/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const reconcileInterval = time.Minute * 3
+
+// newExternalDnsReconciler creates a reconciler that manages external dns resources
+func newExternalDnsReconciler(manager ctrl.Manager, resources []client.Object) error {
+	return common.NewResourceReconciler(manager, "externalDnsReconciler", resources, reconcileInterval)
+}
+
+// NewExternalDns starts all resources required for external dns
+func NewExternalDns(manager ctrl.Manager, conf *config.Config, self *appsv1.Deployment) error {
+	if conf.DNSZoneDomain == "" {
```

Comment:
Is this currently the only thing we need to check?

Reply:
Yes. It matches how WAR currently works.

We could validate that tenantId, subscription, resource group, domain, etc are valid in the config validate function but currently RP is responsible for that before they pass that information to us.

Anything you think we should check?

Reply:
If its settled in RP then we should be good!